### PR TITLE
prometheus connectivity

### PR DESCRIPTION
**Pull request by subhamkrai**
Signed-off-by: subhamkrai <subhamkumarrai03@gmail.com>

### CODE REVIEW

File: handlers/prometheus_handlers.go [hunk 1]
```diff
@@ -51,6 +51,35 @@ func (h *Handler) PrometheusConfigHandler(w http.ResponseWriter, req *http.Reque
 	_, _ = w.Write([]byte("{}"))
 }
 
+// PrometheusPingHandler - fetches server version to simulate ping
```
Comment (girishranganathan) [inline handlers/prometheus_handlers.go]: @subhamkrai this handler code looks good 👍 💯

File: models/handlers.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ type HandlerInterface interface {
 	GrafanaBoardImportForPrometheusHandler(w http.ResponseWriter, req *http.Request, session *sessions.Session, prefObj *Preference, user *User)
 	PrometheusQueryHandler(w http.ResponseWriter, req *http.Request, session *sessions.Session, prefObj *Preference, user *User)
 	PrometheusQueryRangeHandler(w http.ResponseWriter, req *http.Request, session *sessions.Session, prefObj *Preference, user *User)
+	PrometheusPingHandler(w http.ResponseWriter, req *http.Request, session *sessions.Session, prefObj *Preference, user *User)
```
Comment (girishranganathan) [inline models/handlers.go]: perfect

File: router/server.go [hunk 3]
```diff
@@ -44,6 +44,7 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 	mux.Handle("/api/prometheus/board_import", h.AuthMiddleware(h.SessionInjectorMiddleware(h.GrafanaBoardImportForPrometheusHandler)))
 	mux.Handle("/api/prometheus/query", h.AuthMiddleware(h.SessionInjectorMiddleware(h.PrometheusQueryHandler)))
 	mux.Handle("/api/prometheus/query_range", h.AuthMiddleware(h.SessionInjectorMiddleware(h.PrometheusQueryRangeHandler)))
+	mux.Handle("/api/prometheus/ping", h.AuthMiddleware(h.SessionInjectorMiddleware(h.PrometheusPingHandler)))
```
Comment (girishranganathan) [inline router/server.go]: perfect again

File: ui/components/DashboardComponent.js [hunk 4]
```diff
@@ -233,6 +233,33 @@ class DashboardComponent extends React.Component {
     }, self.handleError("Could not ping Kubernetes."));
   }
 
+  handlePrometheusClick = () => {
```
Comment (girishranganathan) [inline ui/components/DashboardComponent.js]: looks great 🏆

File: ui/components/DashboardComponent.js [hunk 5]
```diff
@@ -376,6 +403,7 @@ class DashboardComponent extends React.Component {
           label={prometheus.prometheusURL}
           // onDelete={self.handleDelete}
           // deleteIcon={<DoneIcon />}
+          onClick={self.handlePrometheusClick}
```
Comment (girishranganathan) [inline ui/components/DashboardComponent.js]: 👍

File: ui/components/PrometheusComponent.js [hunk 6]
```diff
@@ -202,6 +202,33 @@ class PrometheusComponent extends Component {
         }, self.handleError);
     }
 
+    handlePrometheusClick = () => {
```
Comment (girishranganathan) [inline ui/components/PrometheusComponent.js]: 👍

File: ui/components/PrometheusSelectionComponent.js [hunk 7]
```diff
@@ -271,7 +271,7 @@ class PrometheusSelectionComponent extends Component {
     render = () => {
         var self = this;
```
Comment (girishranganathan) [inline ui/components/PrometheusSelectionComponent.js]: 👍

**Comment by leecalcote** [comment]
It would be nice if the endpoint field would strip off any extraneous portion of the URL beyond the port number (e.g. `http://kubernetes.docker.internal:32291/graph` - strip off `/graph`), so that the user could enter this URL without having to be premising in how they point to Prometheus.

**Comment by subhamkrai** [comment]
> It would be nice if the endpoint field would strip off any extraneous portion of the URL beyond the port number (e.g. `http://kubernetes.docker.internal:32291/graph` - strip off `/graph`), so that the user could enter this URL without having to be premising in how they point to Prometheus.
> 
> @subhamkrai you might take this as a new issue; as an enhancement to both the Prometheus and Grafana connections.

I'll definitely look to that and will open issue

**Comment by leecalcote** [comment]
@subhamkrai, perfect. Thank you.

**Pull request merged**